crossdock/client/errorstchout/behavior.go

```diff
@@ -0,0 +1,171 @@
+			},
+		},
+		{
+			procedure: "no-such-procedure",
+			body:      []byte{},
+			validate: func(res3 []byte, isAppErr bool, err error) {
+				assert.False(isAppErr, "no-such-procedure must not produce application error")
+				err, ok := err.(tchannel.SystemError)
+				assert.True(ok, "no-such-procedure must produce  system error")
+				if !ok {
+					return
+				}
+				code := tchannel.GetSystemErrorCode(err)
+				assert.Equal(code, tchannel.ErrCodeBadRequest, "must produce bad request error")
+				assert.True(strings.Contains(err.Error(), `unrecognized procedure "no-such-procedure"`))
+			},
+		},
+		{
+			procedure: "bad-response",
+			body:      []byte("{}"),
+			validate: func(res3 []byte, isAppErr bool, err error) {
+				assert.False(isAppErr, "bad-response must not produce an application error")
+				err, ok := err.(tchannel.SystemError)
+				assert.True(ok, "bad-response must produce system error")
+				if !ok {
+					return
+				}
+				code := tchannel.GetSystemErrorCode(err)
+				assert.Equal(code, tchannel.ErrCodeUnexpected, "bad-response must produce unexpected error")
+				assert.True(strings.Contains(err.Error(), `failed to encode "json"`), `got %s`, err.Error())
+			},
+		},
+		{
+			procedure: "unexpected-error",
+			body:      []byte("{}"),
+			validate: func(res3 []byte, isAppErr bool, err error) {
+				assert.False(isAppErr, "unexpected-error procedure must not produce application error")
+				err, ok := err.(tchannel.SystemError)
+				assert.True(ok, "unexpected-error procedure must produce system error")
+				code := tchannel.GetSystemErrorCode(err)
+				assert.Equal(code, tchannel.ErrCodeUnexpected, "must produce transport error")
+			},
+		},
+	}
+
+	server := t.Param(params.Server)
+	serverHostPort := fmt.Sprintf("%v:%v", server, serverPort)
+
+	ch, err := tchannel.NewChannel(serviceName, nil)
+	fatals.NoError(err, "Could not create channel")
+
+	peer := ch.Peers().Add(serverHostPort)
+
+	runTest := func(t test) {
+		var req2, res2, res3 []byte
+		var headers map[string]string
+
+		ctx, cancel := json.NewContext(time.Second)
+		defer cancel()
+
+		ctx = json.WithHeaders(ctx, headers)
+
+		as := "json"
+		call, err := peer.BeginCall(ctx, serviceName, t.procedure, &tchannel.CallOptions{Format: tchannel.Format(as)})
+		fatals.NoError(err, "Could not begin call")
+
+		err = tchannel.NewArgWriter(call.Arg2Writer()).Write(req2)
+		fatals.NoError(err, "Could not write request headers")
+
+		err = tchannel.NewArgWriter(call.Arg3Writer()).Write(t.body)
+		fatals.NoError(err, "Could not write request body")
+
+		err = tchannel.NewArgReader(call.Response().Arg2Reader()).Read(&res2)
+		if err != nil {
+			t.validate(res3, false, err)
+			return
+		}
+
+		isAppErr := call.Response().ApplicationError()
+
+		err = tchannel.NewArgReader(call.Response().Arg3Reader()).Read(&res3)
+		if err != nil {
+			t.validate(res3, isAppErr, err)
+			return
+		}
+	}
+
+	for _, t := range tt {
+		runTest(t)
```

Comment:
Are you trying to reduce levels of indentation? Because it's totally okay to do,

```lang=go
for _, tt := range tests {
  func() {
    defer foo()
    ...
  }()
}
```